codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java

```diff
@@ -238,82 +172,26 @@ public T getAndRemove(T name, T defaultValue) {
     @Override
     public List<T> getAll(T name) {
         checkNotNull(name, "name");
-        List<T> values = new ArrayList<T>(4);
-        int h = hashCodeGenerator.generateHashCode(name);
+
+        LinkedList<T> values = new LinkedList<T>();
```

Comment:
We could do this....however lets handle this is a followup PR if necessary.

If we assume the majority of time this is called it will be on non-empty headers then the check may be just extra overhead.  I would like to see if there is a strong use case for this before adding the check.